src/App.tsx

```diff
@@ -1,11 +1,44 @@
+import { useState } from "react";
+
+import { DisplayCard } from "./components/DisplayCard/DisplayCard";
+import { fetchSingleCharacter } from "./utils/fetchFunctions";
+
 const App = () => {
-	return (
-		<>
-			<div>
-				<h1>Pokémon Battle</h1>
-			</div>
-		</>
-	);
+  const [nameInputValue, setNameInputValue] = useState<string>("");
+  const [entities, setEntities] = useState<any[]>([]);
+
+  const handleSubmit = async () => {
+    setEntities([...entities, await fetchSingleCharacter(nameInputValue)]);
+    setNameInputValue("");
+  };
+
+  return (
+    <>
+      <div>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            handleSubmit();
+          }}
+        >
+          <input
+            value={nameInputValue}
+            type="text"
+            placeholder="Type a name..."
+            onChange={(e) => {
```

Comment:
You should also be able to get rid of this anonymous function.

Given all the other comments here, you should be able to have something like:

```ts
 const handleNameInput = (event: ChangeEvent<HTMLInputElement>) => {
      setNameInputValue(event.currentTarget.value)
  }

  const handleSubmit = async (event: FormEvent) => {
      event.preventDefault()
    setEntities([...entities, await fetchSingleCharacter(nameInputValue)]);
    setNameInputValue("");
  };
```

and the JSX looking like:

```jsx
    <form
        onSubmit={handleSubmit}
    >
      <input
         value={nameInputValue}
         type="text"
         placeholder="Type a name..."
         onChange={handleNameInput}
      />
      <button>Click me</button>
    </form>
```